### ISSUE TITLE
Skin themes styling and customisability

### ISSUE THREAD

**Issue by codyn329** [codyn329/Material#6]
This issue elaborates on task 21 in issue #2. 

Since there's lots of elements, classes and IDs, it could be definitely hard to style the skin manually. So, the idea created was that there should be pre-defined functions to help ease this process for the user. Here are some ideas below!

# JavaScript object
This seems pretty nice. Easy to read syntax, and *should* be user friendly (haven't tested this to a wide audience yet actually, so this is a more of a guess) Here's an example of how it could look: 
```javascript
var skinTheme = {
    primaryColor: 'orange-500',
    secondaryColor: 'blue-500',
    tertiaryColor: 'light-blue-500' 
}
``` 
Or say you want to choose a monochromatic-based color theme! Just enter one of the color keywords from the palette ( source: http://www.materialui.co/colors ), then everything would be automatically stylized for you. 
```javascript
var skinTheme = {
     monochromeTheme: 'amber'
}
```
Thoughts on this idea?

**Comment by georgebarnick** [comment]
JavaScript triggers as the document loads, unlike CSS which triggers simultaneously with the document. When using JavaScript for an interface, it will take a moment (depending on resource load times) to actually show. With CSS, the styles will load immediately. A good example of JavaScript applying styles to a skin is http://greatballcontraption.com, which loads its CSS through local wiki JavaScript currently. For a split second, you see an unstyled Refreshed skin before you see the custom styles. I would advise against this.

**Comment by codyn329** [comment]
Yeah, unwanted flashes aren't a thumbs-up. The other idea could be a dynamic generator that gives a stylesheet based on the colors chosen, so they can put it into MediaWiki:Material.css / User:<username>/material.css, however this would be a whole new project itself. :P Do you have any ideas?
